examples/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright 2021, Intel Corporation
+#
+
+# add compiler flags using macro defined in functions.cmake file
+add_flag(-Wall)
+add_flag(-Wpointer-arith)
+add_flag(-Wsign-compare)
+add_flag(-Wunreachable-code-return)
+add_flag(-Wmissing-variable-declarations)
+add_flag(-fno-common)
+add_flag(-Wunused-macros)
+add_flag(-Wsign-conversion)
+
+add_flag(-ggdb DEBUG)
+add_flag(-DDEBUG DEBUG)
+
+add_flag("-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2" RELEASE)
+
+add_cstyle(examples-all ${CMAKE_CURRENT_SOURCE_DIR}/*/*.[ch])
+add_check_whitespace(examples-all
+		${CMAKE_CURRENT_SOURCE_DIR}/*/*.[ch]
```

Comment:
what is CMAKE_CURRENT_SOURCE_DIR here?
are both  ${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt and ${CMAKE_CURRENT_SOURCE_DIR}/*/CMakeLists.txt needed? if yes, why?

Reply:
CMAKE_CURRENT_SOURCE_DIR here is a `/libuasync/examples` directory. I've added both of them so that `check-whitespace` would check the CMakeLists.txt file from the `examples` directory as well as all the CMakeLists.txt files from all of the directories inside `examples` directory.

Reply:
Since, after changes,  there's only one `CMakeLists.txt` file for examples, I've deleted the ${CMAKE_CURRENT_SOURCE_DIR}/*/CMakeLists.txt from here.